### DISCUSSION TITLE
Use MediaCodec API for Android

### DISCUSSION THREAD

**Discussion by kairusds** [ppy/osu#22294] [Ideas] [answered]
I think this is a better option than using ffmpeg just to reduce the overhead and improve performance as it actually uses the built-in low-level APIs.

I thought I'd just leave this suggestion here in case the development shifts focus on this specific os.

**Comment by bdach** [comment, marked as answer]
FFmpeg has the benefit of just-working on all platforms and not needing to go out of our way to support Android.

Gains would have to be demonstrably significant to even begin considering this.

**Reply by kairusds** [reply]
I'm not sure about the performance since I haven't really used the API in any real-world projects yet but it has the ability to encode videos from a Surface class which might be useful for turning replays into videos without the need for using screen recording software.

**Reply by bdach** [reply]
Recording replays into videos is out of scope of the project, sorry.

**Comment by tutacat** [comment]
FFMpeg/libavcodec supports MediaCodec already, you just need to select it as the hwaccel (decoder). Video currently causes too much latency to be used realistically.

**Reply by peppy** [reply]
cc @FreezyLemon for visibility

**Reply by FreezyLemon** [reply]
Not very familiar with MediaCodec, but it looks simple enough, assuming that enabling the hwaccel in FFmpeg is enough.

EDIT: Looks like MediaCodec is exposed through separate decoders (e.g. `h264_mediacodec` instead of `h264`). Needs a bit of special handling, but still doable.
